### ISSUE TITLE
p[trace|metric|log]otlp.Response and p[trace|metric|log]otlp.Request do not follow naming convention

### ISSUE THREAD

**Issue by bogdandrutu** [open-telemetry/opentelemetry-collector#6365]
Should they be named "Export[Logs|Metrics|Trace]ServiceRequest"?

/cc @dmitryax

**Comment by bogdandrutu** [comment]
Another option is to drop the prefix `Export[]`, if we do that we should remove it from `Export[Logs|Metrics|Traces]PartialSuccess`

**Comment by dmitryax** [comment]
`Export[Logs|Metrics|Trace]ServiceRequest` maybe seemed like unnecessary long, so we potentially can add it to the exception list, but I think `Export` part is important enough to keep.

Given that these structs are used pretty rarely, I'd rather follow the guidelines here and apply the full name. WDYT?

**Comment by dmitryax** [comment]
or maybe `p[trace|metric|log]otlp.Export[Request|Response]` and `p[trace|metric|log]otlp.ExportPartialSuccess`?

**Comment by bogdandrutu** [comment]
I like the idea of dropping the data type from the name, since it is in the package name anyway. So I think we can add that as a guidance that, when the data type is in the name and does not provide value (like `ResourceLogs`) we should drop it.

So, we will have:

`p[trace|metric|log]otlp.ExportService[Request|Response]`
`p[trace|metric|log]otlp.ExportPartialSuccess`

**Comment by dmitryax** [comment]
Makes sense, but I'm a bit unsure about `ExportService[Request|Response] `. It sounds a bit confusing comparing to `Export[Request|Response]`. Not sure why `ExportMetricsServiceRequest` message was introduced in proto, I believe it should be `ExportRequest` based on examples from https://developers.google.com/protocol-buffers/docs/proto3#services

**Comment by bogdandrutu** [comment]
You should also post this on the proto repo. I think you are right.
